From 9573bd5bd26774f482d85a413bc456b6e54204bd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C4=90o=C3=A0n=20Tr=E1=BA=A7n=20C=C3=B4ng=20Danh?=
 <congdanhqx@gmail.com>
Date: Sat, 31 Oct 2020 19:45:00 +0700
Subject: [PATCH] android-base/endian.h: fix build on musl

So far musl has used the definitions for Windows. Now that it includes
<winsock2.h> this fails horribly. Add a proper fallback for !__GLIBC__
but __linux__ with defines for musl.
---
 include/android-base/endian.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/include/android-base/endian.h b/include/android-base/endian.h
index 1e5cceb..7986ec8 100644
--- a/include/android-base/endian.h
+++ b/include/android-base/endian.h
@@ -44,6 +44,18 @@
 #define letoh32(x) le32toh(x)
 #define letoh64(x) le64toh(x)
 
+#elif defined(__linux__)
+
+/* musl's <endian.h> is like bionic's <sys/endian.h>. */
+#include <endian.h>
+
+/* musl keeps htons and htonl in <netinet/in.h>. */
+#include <netinet/in.h>
+
+/* musl doesn't have the 64-bit variants. */
+#define htonq(x) htobe64(x)
+#define ntohq(x) be64toh(x)
+
 #else
 
 #if defined(__APPLE__)
